Challenge-4/Bot_2/src/helper.spec.ts

```diff
@@ -0,0 +1,133 @@
+          supplyPerYearInterestRateBase: ethers.BigNumber.from(900),
+          borrowKink: ethers.BigNumber.from(4000),
+          borrowPerYearInterestRateSlopeLow: ethers.BigNumber.from(100),
+          borrowPerYearInterestRateSlopeHigh: ethers.BigNumber.from(200),
+          borrowPerYearInterestRateBase: ethers.BigNumber.from(300),
+          storeFrontPriceFactor: ethers.BigNumber.from(50),
+          trackingIndexScale: ethers.BigNumber.from(50),
+          baseTrackingSupplySpeed: ethers.BigNumber.from(50),
+          baseTrackingBorrowSpeed: ethers.BigNumber.from(50),
+          baseMinForRewards: ethers.BigNumber.from(50),
+          baseBorrowMin: ethers.BigNumber.from(50),
+          targetReserves: ethers.BigNumber.from(50),
+          assetConfigs: [
+            {
+              asset: usdcTokenAddress,
+              decimals: 18,
+              conversionFactor: ethers.BigNumber.from(1),
+            },
+          ],
+        },
+      ],
+    })
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getUtilization", {
+      inputs: [],
+      outputs: [ethers.BigNumber.from(5000)],
+    })
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getSupplyRate", {
+      inputs: [ethers.BigNumber.from(5000)],
+      outputs: [ethers.BigNumber.from(500000000)],
+    });
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getBorrowRate", {
+      inputs: [ethers.BigNumber.from(5000)],
+      outputs: [ethers.BigNumber.from(500000000)],
+    });
+
+    const getConfig: any = await helper.getAllCompoundData(
+      usdcTokenAddress,
+      mockConfigProxy,
+      0,
+    );
+    
+    const expectedConfig = [
+      createAddress("0x123"),
+      createAddress("0x123"),
+      createAddress("0x123"),
+      createAddress("0x123"),
+      createAddress("0x223"),
+      ethers.BigNumber.from(5000).toString(),
+      ethers.BigNumber.from(60).toString(),
+      ethers.BigNumber.from(100).toString(),
+      ethers.BigNumber.from(900).toString(),
+      ethers.BigNumber.from(4000).toString(),
+      ethers.BigNumber.from(100).toString(),
+      ethers.BigNumber.from(200).toString(),
+      ethers.BigNumber.from(300).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      ethers.BigNumber.from(50).toString(),
+      [[usdcTokenAddress, 18, ethers.BigNumber.from(1)]],
+    ];
+
+    const getConfigStringified = getConfig.configurationData.map((item: any) => {
+      if (Array.isArray(item)) {
+        return item.map((subItem) =>
+          ethers.BigNumber.isBigNumber(subItem) ? subItem.toString() : subItem,
+        );
+      }
+      return ethers.BigNumber.isBigNumber(item) ? item.toString() : item;
+    });
+    const stringifiedConfigDataStringField = JSON.stringify(getConfigStringified);
```

Comment:
Couldn't you add a step like this by chaining to `.map()`, since that returns an array? Then you don't have to create a separate line to declare another variable.